packages/lesswrong/components/hooks/useCrosspostApolloClient.tsx

```diff
@@ -0,0 +1,12 @@
+import type { ApolloClient, NormalizedCacheObject } from '@apollo/client';
+import { createApolloClient } from "../../client/apolloClient";
+import { fmCrosspostBaseUrlSetting } from "../../lib/instanceSettings";
+
+let apolloClient: ApolloClient<NormalizedCacheObject> | undefined;
+
+export const useCrosspostApolloClient = () => {
+  if (!apolloClient) {
+    apolloClient = createApolloClient(fmCrosspostBaseUrlSetting.get() ?? undefined);
```

Comment:
I'm not sure the optimization is worth it - if we recreate a new client each time, are we losing much?  It's not a super expensive operation and I don't expect there to be any benefit from the graphql caching in this case.

Reply:
I can see your point, but memoizing the value only adds 2 lines of code and is completely invisible from outside this file so I don't think there's really any cost to doing so.

Reply:
The main reason I suggested it here is that the apollo client is a suspect in our memory leak, and they [explicitly recommend](https://github.com/apollographql/apollo-client/issues/7942#issuecomment-812540838) making a new client instance on each request when using the client in an SSR context.  Normally I'd be all for treating things like clients as singletons but the apollo client in particular seems like a pretty risky place to do so.

Reply:
Ah, in that case it makes sense! Sorry - wasn't aware that this was the cause of the memory leak.